ygen/enumgen.go

```diff
@@ -103,10 +104,14 @@ func (s *enumSet) enumeratedUnionEntry(e *yang.Entry, compressPaths, noUnderscor
 		case t.Enum != nil:
 			var enumName string
 			var err error
-			if util.IsYANGBaseType(t) {
+			enumNameSake := t
+			if useDefiningModuleForTypedefEnumNames {
+				enumNameSake = subsumingTypes[i]
```

Comment:
It wasn't clear to me why this gives us the right 'subsuming' type -- why is the index here of `EnumeratedUnionTypes` the right index to use in `EnumeratedUnionSubsumingTypes`, since AIUI, if we have a `union` type it'll be returned more than once (see comments on `parent-type` in the other change), so I don't think that I understand why there would be a 1:1 correspondence between the two.

I wonder whether it's best to have something whereby we do:

`typeTree := buildTypeTree(e.Type.Type)`

and then later:
 
`subsumingType := typeTree.nonEnumeratedParentTypeFor(type)`

This would remove there being any dependency on ordering.

Reply:
Now that `DefiningType` is being used, hopefully this looks much cleaner.